test/suite/test_compact10.py

```diff
@@ -0,0 +1,146 @@
+            c.set_key(i)
+            c.remove()
+        c.close()
+
+    def get_bg_compaction_running(self):
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        stat_cursor.close()
+        return compact_running
+
+    def get_pages_rewritten(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, None)
+        pages_rewritten = stat_cursor[stat.dsrc.btree_compact_pages_rewritten][2]
+        stat_cursor.close()
+        return pages_rewritten
+
+    def populate(self, uri, num_keys, value_size):
+        c = self.session.open_cursor(uri, None)
+        for k in range(num_keys):
+            c[k] = ('%07d' % k) + '_' + 'abcd' * ((value_size // 4) - 2)
+        c.close()
+
+    def turn_on_bg_compact(self, config):
+        self.session.compact(None, config)
+        compact_running = self.get_bg_compaction_running()
+        while not compact_running:
+            time.sleep(1)
+            compact_running = self.get_bg_compaction_running()
+        self.assertEqual(compact_running, 1)
+
+    def test_compact10(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create and populate tables.
+        uris = []
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            self.session.create(uri, self.create_params)
+            self.populate(uri, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of the file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Reconfigure the connection with:
+        # - Aggressive compaction
+        reconf_config = self.conn_config + ',debug_mode=(background_compact)'
+        # - Stressing options to slow down compaction and checkpoints so the test can trigger the
+        # crash while compaction is still in progress.
+        reconf_config += f',timing_stress_for_test=({self.timing_stress_cfg})'
+        # - Verbose (for debugging purposes)
+        # reconf_config += ',verbose=(compact,compact_progress)'
+        self.conn.reconfigure(reconf_config)
+        
+        # Enable background compaction. Exclude the HS file as there is no content to compact.
+        exclude_list = '["table:WiredTigerHS.wt"]'
+        compact_config = f'background=true,free_space_target=1MB,exclude={exclude_list}'
+        self.turn_on_bg_compact(compact_config)
+
+        # Wait for a random table to be compacted before crashing.
+        uri_idx = random.randint(0, self.n_tables - 1)
+        uri = uris[uri_idx]
+        while not self.get_pages_rewritten(uri):
+            time.sleep(0.5)
+
```

Comment:
At this point we're sure compaction has started to do work. Maybe we could add a random amount of time to sleep before crashing to try and crash at different points during compaction.

Reply:
Agreed, see ff5606152fac4acf0ee6e6d12ef8b8ba0275bf3e